app/pokemon/page.tsx

```diff
@@ -1,43 +1,81 @@
-"use client"
+'use client'
 
-import { useEffect, useState } from "react"
-import Link from "next/link"
-import { get } from "@/lib/request"
+import Link from 'next/link'
+import { get } from '@/lib/request'
+import axios from 'axios'
+import { useEffect, useState } from 'react'
 
 type Pokemon_species = {
-    name: string
-    url: string
+  name: string
+  url: string
 }
 
-
 type Pokedex_Entry = {
-    entry_number: number
-    pokemon_species: Pokemon_species
+  entry_number: number
+  pokemon_species: Pokemon_species
 }
 
 // gets all the pokemon in a list from our secret api and returns them so you can view each on a new page
 export default function Page() {
-    const [pokemon, setPokemon] = useState<any[]>([])
-
-    useEffect(() => {
-        get(`${process.env.NEXT_PUBLIC_SECRET_API!}`)
-            .then(res => setPokemon(res.data.pokemon_entries))
-
-        console.log(pokemon)
-    }, [])
-
-    return (
-        <section>
-            <h2>Check out the full list of pokemon here!</h2>
-            {pokemon && pokemon.map((poke: Pokedex_Entry) => {
-                return (
-                    <>
-                        <Link href={poke.pokemon_species.url}>
-                            <h4>{poke.pokemon_species.name}</h4>
-                        </Link>
-                    </>
-                )
-            })}
-        </section>
-    )
+  const [pokemon, setPokemon] = useState<Pokedex_Entry[]>([])
+  const [sortBy, setSortBy] = useState('alphabet')
+  const [sortedPokemon, setSortedPokemon] = useState<Pokedex_Entry[]>([])
+
+  useEffect(() => {
+    get('/api/pokemon').then(({ data: { pokemon } }) => setPokemon(pokemon))
+  }, [])
```

Comment:
cool so this is how things used to be done before server components. This will work but it exposes the api content to the client making it very hard to protect your api.